### ISSUE TITLE
A few torrents are not starting at all

### ISSUE THREAD

**Issue by carlonluca** [ygrek/mldonkey#22]
I noticed most torrents start properly. Some others are not starting at all. For instance I found this magnet link which properly downloads with other clients but not with mldonkey.

`magnet:?xt=urn:btih:BB1A0A400441F897E86ABA05305447AC55591CCA&dn=Dr.House+Stagione+6+XviD.torrent&tr=http%3A%2F%2Ftracker.tntvillage.scambioetico.org%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.tntvillage.scambioetico.org%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969%2Fannounce&tr=udp%3A%2F%2FIPv6.leechers-paradise.org%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.internetwarriors.net%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.tiny-vps.com%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.mg64.net%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80%2Fannounce`

Can you reproduce this problem? Some others are behaving like this actually, even though most torrents start properly.

**Comment by uzerus** [comment]
yeah, i have this issue too, with torrent that have over 1000 peers...  
OS: ubuntu 18.04
MLDonkey: latest stable

**Comment by alvarogonzalezsotillo** [comment]
I can reproduce this problem  with udp only trackers.  It seems that udp trackers are not supported

**Comment by carlonluca** [comment]
HTTPS trackers were also not implemented. My fork includes support for HTTPS. Not sure about udp.

**Comment by carlonluca** [comment]
> I can reproduce this problem with udp only trackers. It seems that udp trackers are not supported

I see an implementation of UDP trackers: https://github.com/ygrek/mldonkey/blob/master/src/networks/bittorrent/bTUdpTracker.mlp. Are you sure about that? Cause we may look inside that implementation for the problem. Are you able to provide a way to reproduce?

**Comment by alvarogonzalezsotillo** [comment]
I tried:

`magnet:?xt=urn:btih:ASOJ47UJNYIZXIHMD26TQDGEJ4MPSVCL&dn=Fedora-KDE-Desktop-Live-aarch64-43&&tr=udp://tracker.openbittorrent.com:80&tr=udp://tracker.opentrackr.org:1337/announce`

At first, the trackers are shown:

<img width="593" height="704" alt="Image" src="https://github.com/user-attachments/assets/535b1f56-1e83-4a3c-b70d-d824e72f60f8" />

But then, the error `disabled: error: tracker type not supported`

<img width="657" height="777" alt="Image" src="https://github.com/user-attachments/assets/3955943b-37de-4be6-b0a2-919d9be7617f" />

Version is `MLNet Multi-Network p2p client version 3.2.1.git`

Thanks for your time!

**Comment by carlonluca** [comment]
I investigated the issue a bit: I found two bugs by just looking at the code. With the first patch, udp results as a supported tracker protocol (it should be). With the second patch, metadata can be successfully received. After these two patches, I could start torrents that I was never able to start before. If you want to try, I uploaded these two patches to my mldonkey fork: https://github.com/carlonluca/mldonkey.

Unfortunately, I found other bugs. Not sure if it makes sense to go on, I wonder how many problems I may find.

**Comment by alvarogonzalezsotillo** [comment]
I suppose the branch to try is `dev`,  is that correct?

**Comment by carlonluca** [comment]
Correct. The last two patches should fix the first two issues. Something should then start.
